File: app/src/main/java/me/shikhov/wlogtest/MainActivity.java
package me.shikhov.wlogtest;

import android.app.Activity;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import me.shikhov.wlog.Log;


public class MainActivity extends Activity
{
    private static final String TAG = MainActivity.class.getSimpleName();

    class KnownPeople
    {
        class Person
        {
            String name;
            int age;

            @Override
            public String toString()
            {
                return super.toString();
            }
        }

        List<Person> schoolFriends = new ArrayList<>();

        Map<String,Person> universityFriends = new HashMap<>();

        @Override
        public String toString()
        {


            return super.toString();
        }
    }


    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        Log.get(TAG).event(this, "onCreate").i().release();

        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume()
    {
        super.onResume();
        Log.get(TAG).event(this, "onResume")
                .f()
                .a(new Integer[]{1,2,3,4,5})
                .release();

        List<String> list = new ArrayList<>();
        list.add("one");
        list.add("two");
        list.add("three");

        Log.get(TAG).a(list).r();

        HashMap<String,String> map = new HashMap<>();
        map.put("one", "not two");
        map.put("two", "not three");
        map.put("three", "not four");
        map.put("four", null);

        Log.get(TAG).a(map).r();

        List<String> t = new LinkedList<>();
        t.add("singleItem");

        Map<String, Integer> m = new LinkedHashMap<>();
        m.put("singleItem", 8);

        Log.get(TAG).a(t).r();
        Log.get(TAG).a(m).r();
    }

    @Override
    protected void onPause()
    {
        super.onPause();
        Log.get(TAG).event(this, "onPause").release();
    }

    @Override
    protected void onDestroy()
    {
        super.onDestroy();
        Log.get(TAG).event(this, "onDestroy").release();
    }
}

File: app/src/main/java/me/shikhov/wlogtest/MainActivity.kt
package me.shikhov.wlogtest

import android.app.Activity
import android.os.Bundle
import me.shikhov.wlog.Log
import java.net.UnknownServiceException
import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.HashMap
import kotlin.collections.LinkedHashMap
import kotlin.concurrent.thread

class MainActivity : Activity() {

    inner class KnownPeople {

        inner class Person {
            var name: String? = null
            var age = 0
            override fun toString(): String {
                return super.toString()
            }
        }

        val schoolFriends: MutableList<Person> = ArrayList()

        var universityFriends: Map<String, Person> = HashMap()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log(TAG).event(this, "onCreate").r()
        setContentView(R.layout.activity_main)

        thread {
            Log("THREAD").event(this, "onStart").r()
            Log("THREAD", logLevel = Log.WARN) {
                line(nlBefore = true)
                +"TEXT"
                "PAIR_KEY".."PAIR_VALUE"
                "PAIR_KEY2"..15
                line('*', nlBefore = true)
            }
            Log("THREAD").event(this, "onEnd").r()
        }
    }

    val someProp = "hello!"

    override fun onResume() {
        super.onResume()
        Log(TAG).event(this, "onResume").r()

        Log(TAG).a(arrayOf(1, 2, 3, 4, 5)).r()

        val list: MutableList<String> = ArrayList()
        list += "one"
        list += "two"
        list += "three"

        Log(TAG).a(list).r()

        val map = HashMap<String, String?>()
        map["one"] = "not two"
        map["two"] = "not three"
        map["three"] = "not four"
        map["four"] = null
        Log(TAG).a(map).r()

        val t = LinkedList<String>()
        t.add("singleItem")
        val m: MutableMap<String, Int> = LinkedHashMap()
        m["singleItem"] = 8
        Log(TAG).a(t).r()
        Log(TAG).a(m).r()
        Log(TAG).a("<< arrays >>").r()
        val testArr = intArrayOf(0, 1, 2, 3, 5, 8, 13, 21)
        Log(TAG).a("array: ").a(testArr).r(Log.WARN)

        Log(TAG) {
            line()
            throwable = UnknownServiceException()
            +"test message 1"
            +map
            +testArr
            +"test message 2"
            +::someProp
            line(char = '*')
        }

        Log(TAG).a("property: ").a(::someProp).r(Log.INFO)

        Log(TAG).a("onResume end").r()
    }

    override fun onPause() {
        super.onPause()
        Log(TAG).event(this, "onPause").r()
    }

    override fun onDestroy() {
        super.onDestroy()
        Log(TAG).event(this, "onDestroy").r()
    }

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }
}